asmcomp/amd64/emit.mlp

```diff
@@ -748,6 +748,11 @@ let emit_instr fallthrough i =
       I.sqrtsd (arg i 0) (res i 0)
   | Lop(Ispecific(Ifloatsqrtf addr)) ->
       I.sqrtsd (addressing addr REAL8 i 0) (res i 0)
+  | Lop(Ispecific(Isextend)) ->
+      begin match (arg i 0, res i 0) with
+      | (Reg64 RAX, Reg64 RAX) -> I.cdqe ()
+      | (_, dst)               -> I.movsxd (arg32 i 0) dst
```

Comment:
I'm skeptical that it is worth recognizing and generating the `cdqe` special-case instruction.  Emitting `movsxd` in all cases would be just as efficient in practice and keep the code simpler.

Reply:
Roger that, I'll update the PR as soon as possible.